### DISCUSSION TITLE
Constructing this " nlohmann::json jsondata{{{"domain","test"}}};  " creates an json array

### DISCUSSION THREAD

**Discussion by danielmevi** [nlohmann/json#2391] [Q&A] [answered]
I prefer to construct with `{}` instead of `=`.
Then when trying to swap the `=` with a pair of `{}` to one of the examples from documentation 
```
json j2 = {
  {"pi", 3.141},
  {"happy", true},
  {"name", "Niels"},
  {"nothing", nullptr},
  {"answer", {
    {"everything", 42}
  }},
  {"list", {1, 0, 2}},
  {"object", {
    {"currency", "USD"},
    {"value", 42.99}
  }}
};
```

into a `{}` constructor, which ended up in:
```
json j2 { 
{
  {"pi", 3.141},
  {"happy", true},
  {"name", "Niels"},
  {"nothing", nullptr},
  {"answer", {
    {"everything", 42}
  }},
  {"list", {1, 0, 2}},
  {"object", {
    {"currency", "USD"},
    {"value", 42.99}
  }}
}
};
```

The run ended up with exception when trying to insert a new key `j2["name"] = "Niels";`
```
terminate called after throwing an instance of 'nlohmann::detail::type_error'
  what():  [json.exception.type_error.305] cannot use operator[] with a string argument with array
```

After some time I made it work by removing one of the 3 pairs of `{}` from the constructor. I am deducing that having 3 pairs of `{}` creates an array, which does not have the `operator[]` for it.

am I missing something? otherwise it would be good that documentation does mention it or that such constructor becomes invalid like std vector rejects a `std::vector<int> vec{{{1,2}}};` with a `<brace-enclosed initializer list>` compilation error

**Comment by nlohmann** [comment, marked as answer]
This seems to be related to https://github.com/nlohmann/json/issues/2311.
